node/src/components/block_validator.rs

```diff
@@ -84,6 +85,19 @@ pub(crate) enum Event {
     #[from]
     Request(BlockValidationRequest),
 
+    /// Past blocks with metadata relevant to the finality signatures included in the proposed
+    /// block have been read from storage.
+    #[display(fmt = "{:?} read from storage", past_blocks_with_metadata)]
```

Comment:
This could potentially create a huge message, ain't it? Is the size of this vector constrained?

Reply:
It should be at most `signature_rewards_max_delay` blocks, which will most likely always be just a few (temporarily set to 3 at the moment). But since it technically can be set to an arbitrary value, it might be a good idea to change the message here...